gnucash/import-export/ofx/gnc-ofx-import.cpp

```diff
@@ -1278,49 +1284,48 @@ reconcile_when_close_toggled_cb (GtkToggleButton *togglebutton, ofx_info* info)
     info->run_reconcile = gtk_toggle_button_get_active (togglebutton);
 }
 
-static gchar* make_date_amount_key (time64 date, gnc_numeric amount)
+static const std::string
+make_date_amount_key (time64 date, gnc_numeric amount)
 {
     // Create a string that combines date and amount, we'll use that for our hash
     gchar buf[64];
     gnc_numeric _amount = gnc_numeric_reduce(amount);
-    g_snprintf (buf, sizeof(buf), "%" PRId64 "%" PRId64 "%" PRId64, _amount.num , _amount.denom, date);
-    return g_strdup (buf);
+    g_snprintf (buf, sizeof(buf), "%s%" PRId64 "%" PRId64, GncDateTime(date).format("%y%m%d").c_str(), _amount.num , _amount.denom);
```

Comment:
this "%y%m%d" format ensure the resulting `buf` together with the num+denom is typically 15 bytes or less, ensuring SSO is done in both gcc and clang...

Reply:
Typically in rich countries. 6 bytes for the date leaves 9; if denom is 100 that leaves only 6 for num, so any amount over 9999.99 overflows. That won't even buy a cup of coffee in inflated currencies like VND or COP.

If you want to save allocations create a stack `buf` in the calling function, wrap it in a std::string_view and pass the std::string_view& into `make_date_amount_key`.  That's safe to pass to `trans_hash.find` but not to store anywhere.